src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -1496,4 +1497,19 @@ public TopicChange publishTopic(TopicChange topicChange) {
   
   
 
+  @Override
+  public String translate(String text) {
+    Response response = getResponse(text);
+    return response.msg;
+  }
+
+  @Override
+  public String translate(String text, String fromLang, String toLang) {
+    // FIXME implement - same mapping that current inmoov has ? en-US etc ..
+    // although all of those are localizations
+    // not languages only the 1st part is "en" in en-US
+    // return getResponse(text)
+    return null;

```

Comment:
If possible I think a full implementation would be better than a stub returning null, but at the same time I'm not totally sure how to do that with program AB

Reply:
Yeah ... I'll need to address this in the short future ..
As @kwatters mentioned, the translation layer will be implemented as an overlay with key templates and a key driven bot.
I'm still not completely sure if ProgramAB should or should not support translation semantics ..